docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfd-fraud-proofs.md

```diff
@@ -0,0 +1,173 @@
+
+![PFB Merkle Proof](./assets/pfd-merkle-proof.png)
+
+Let's assume a square size of k. The amount of blue nodes from the shares to ROW1 is O(log(k). The amount of blue nodes from ROW1 to the `DataRoot` is also O(log(k). You will have to include the shares themselves in the proof.
+Share size := 512 bytes
+NMT-Node size := 32 bytes + 2\*8 bytes = 48 bytes
+MT-Node size := 32 bytes
+
+Worst Case Normal PFB proof size in bytes  
+= 2 PFB Shares + blue nodes to row roots + blue nodes to (`DataRoot`)  
+= 2 \* Share size + 2 \* log(k) \* NMT-Node size + log(k) \* MT-Node size  
+= 2 \* 512 + 2 \* log(k) \* 48 + log(k) \* 32  
+= 1024 + 128 \* log(k)  
+
+As the size of a PFB transaction is unbounded you can encompass even more shares. To put a bound on this we assume that most PFB transactions will be able to be captured by 4 Shares.
+
+Huge PFB proof size in bytes  
+= 4 PFB Shares + blue nodes to row roots + blue nodes to (`DataRoot`)  
+= 4 \* Share size + 2 \* log(k) \* NMT-Node size + log(k) \* MT-Node size  
+= 2048 + 128 \* log(k)
+
+### Blob Inclusion Proof
+
+The worst-case blob inclusion proof size will result from the biggest possible blob with the most amount of subtree roots. This blob is constructed as filling up the whole block and having the last row missing one share.
+
+![Blob Inclusion Proof](./assets/blob-merkle-proof.png)
+
+With a blob of size n and a square size of k, this means that we have O(sqrt(n)) subtree row roots and O(log(k)) subtree row roots in the last row. As the whole block is filled up, sqrt(n) tends towards k. We will also require additional k blue parity nodes to prove the row roots.
+
+Worst case blob inclusion proof size  
+= subtree roots (all rows) + subtree roots (last row) + blue nodes (parity shares)  
+= (sqrt(n) + log(k) + k) \* NMT-Node size     | sqrt(n) => k  
+= ( 2 \* k + log(k) ) \* 48
+
+## Optimizations
+
+### Optimization 1
+
+If a PFB would be guaranteed to be in one share then we could decrease the PFB proof size significantly. You would not only get rid of one share resulting in a 512 bytes save but also log(k) fewer blue nodes in the worst case. The maximum size PFB that fits into one share is 501 bytes long:  
+= Share size - nid - special byte - reserved bytes - transaction length  
+= 512 - 8 - 1 - 1 - 1 = 501  
+Therefore a normal-sized PFB of 330 bytes fits easily into a share with additional spare bytes to be used for more complex PFBs.
+A requirement for this is an option for the transaction to be the only one in a share or at the start of a share if it fits in the share. This would extend to bigger PFB shares as well so you can potentially reduce the size of overlapping PFB proofs by 1 share (512 bytes).
```

Comment:
[no change needed] interesting. This sounds like writing transactions to sparse shares instead of compact shares. Ref: https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#share

Reply:
Hm yes seems like that for optimization 2. Although for Optimization 1 we only need the guarantee that the pfb is in one share. So there could be other transactions present if the pfb starts and ends in the same share